src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,138 @@
+package duke;
+
+import java.util.ArrayList;
+
+import duke.exception.DukeException;
+import duke.exception.InvalidInput;
+import duke.task.Task;
+
+
+/**
+ * A class that encapsulates the storing of all the task data.
+ */
+public class TaskList {
+    /** The ArrayList that contains all the tasks */
+    private final ArrayList<Task> data;
+
+    /**
+     * The class constructor that initialises the ArrayList.
+     */
+    public TaskList() {
+        this.data = new ArrayList<>();
+    }
+
+    /**
+     * Returns the number of tasks saved.
+     *
+     * @return The size of the ArrayList.
+     */
+    public int size() {
+        return data.size();
+    }
+
+    /**
+     * Returns the tasks saved in the provided index.
```

Comment:
Perhaps using `task` instead of `tasks` would be more appropriate? Since this method only returns a single Task object.